lib/waterline/utils/normalize.js

```diff
@@ -50,6 +50,25 @@ var normalize = module.exports = {
       else if (context.attributes[pk].type == 'string') {
         coercePK = function(pk) {return String(pk).toString();};
       }
+
+      // If the the data type is uuid (both according to the model definition
+      // as well as the database schema) attempt to coerce the value to a
+      // properly formatted uuid, stripping any extreneous characters from it.
+      else if (context.attributes[pk].type == 'uuid' &&
```

Comment:
That’d be my normal preference, though I was deferring to the [existing style](https://github.com/Shyp/waterline/pull/32#discussion-diff-63740139R50). I could go either way, but since the RHS is a string literal, I’m not super worried about it.